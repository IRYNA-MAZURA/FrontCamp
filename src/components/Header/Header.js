import React from 'react';
import Logo from "../Logo";
import Search from "../Search";

import './Header.css';

export const Header = () => {
    return (
        <header>
            <Logo/>
            <Search/>
        </header>
    )
};
