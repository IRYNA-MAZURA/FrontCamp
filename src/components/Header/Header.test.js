import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { Header } from './Header';

const mockStore = configureStore([]);
const store = mockStore({
    search:
    {
        searchBy: 'title',
        value: 'Harry Potter'
    }    
});

describe('Header component', () => {
    it('snapshot testing', () => {
        const wrapper = renderer.create(<Provider store={store}><Router><Header /></Router></Provider>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});