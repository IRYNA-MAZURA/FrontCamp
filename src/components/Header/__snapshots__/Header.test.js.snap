// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component snapshot testing 1`] = `
<header>
  <div
    className="logo"
  >
    <a
      className="logo-link"
      href="/"
      onClick={[Function]}
    >
      <strong>
        netflix
      </strong>
      roulete
    </a>
  </div>
  <div
    className="search"
  >
    <div
      className="search-title"
    >
      <h2>
        find your movie
      </h2>
    </div>
    <div
      className="search-text"
    >
      <input
        className="search-text-input"
        defaultValue="Harry Potter"
        name="search"
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
      <a
        href="/title/"
        onClick={[Function]}
      >
        <button
          className="main active"
          onClick={[Function]}
        >
          Search
        </button>
      </a>
    </div>
    <div
      className="search-type"
    >
      <span>
        search by
      </span>
      <button
        className="left active"
        onClick={[Function]}
      >
        title
      </button>
      <button
        className="right"
        onClick={[Function]}
      >
        genre
      </button>
    </div>
  </div>
</header>
`;
